### PR TITLE
refactor: file folders

### PR DESCRIPTION
**Pull request by seavan**
Continue refactor for - https://github.com/PeerioTechnologies/peerio-desktop/pull/123

### CODE REVIEW

File: src/helpers/icebear-dom.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { fileStore } = require('peerio-icebear');
+const { getAttributeInParentChain } = require('./dom');
+
+function getFolderByEvent(ev) {
+    const folderId = getAttributeInParentChain(ev.target, 'data-folderid');
+    // TODO: add unique id for root folder
```
Comment (anri-asaturov) [inline src/helpers/icebear-dom.js]: What do you mean? root is unique enough, no? Our folder ids can never be generated as 'root' randomly.

File: src/stores/ui-store.js [hunk 2]
```diff
@@ -107,5 +103,5 @@ class UIStore {
 
 const store = new UIStore();
 clientApp.uiUserPrefs = store.prefs;
-
+window.uiStore = store;
```
Comment (anri-asaturov) [inline src/stores/ui-store.js]: ?

**Comment by anri-asaturov** [comment]
@seavan: plz fix lint issues.
@ebilg: plz test if this refactor didn't break folders and files section of sidebar.

File: src/helpers/icebear-dom.js (hunk 1 above)
Comment (seavan) [reply src/helpers/icebear-dom.js]: yes, currently folderId of root folder is null (and logic in desktop and mobile depends on it I think, although it should use 'isRoot' property)

we need to change it to 'root' being the folderid instead - will make a lot of things more intuitive.

not in this PR though.

File: src/stores/ui-store.js (hunk 2 above)
Comment (seavan) [reply src/stores/ui-store.js]: :D

**Comment by seavan** [comment]
https://app.clubhouse.io/peerio/story/976/file-folders-refactor-and-pr

**Comment by ebilg** [comment]
tested creating a folder, moving folder, renaming folder, moving file into folder, uploading to folder with picker, with drag and drop, sync between devices and recent files sidebar. looks good

**Comment by anri-asaturov** [comment]
@seavan tests are failing

**Comment by seavan** [comment]
@anri-asaturov now they're not

**Pull request merged**